src/components/Sidebar/useMenuStructure.tsx

```diff
@@ -0,0 +1,292 @@
+import { appsListPath } from "@dashboard/apps/urls";
+import {
+  extensionMountPoints,
+  useExtensions,
+} from "@dashboard/apps/useExtensions";
+import { categoryListUrl } from "@dashboard/categories/urls";
+import { collectionListUrl } from "@dashboard/collections/urls";
+import { MARKETPLACE_URL } from "@dashboard/config";
+import { configurationMenuUrl } from "@dashboard/configuration";
+import { getConfigMenuItemsPermissions } from "@dashboard/configuration/utils";
+import { customerListUrl } from "@dashboard/customers/urls";
+import { saleListUrl, voucherListUrl } from "@dashboard/discounts/urls";
+import { giftCardListUrl } from "@dashboard/giftCards/urls";
+import { PermissionEnum, UserFragment } from "@dashboard/graphql";
+import { commonMessages, sectionNames } from "@dashboard/intl";
+import { marketplaceUrlResolver } from "@dashboard/marketplace/marketplace-url-resolver";
+import { orderDraftListUrl, orderListUrl } from "@dashboard/orders/urls";
+import { pageListPath } from "@dashboard/pages/urls";
+import { productListUrl } from "@dashboard/products/urls";
+import { languageListUrl } from "@dashboard/translations/urls";
+import { SidebarMenuItem } from "@saleor/macaw-ui";
+import {
+  ConfigurationIcon,
+  CustomersIcon,
+  HomeIcon,
+  MarketplaceIcon,
+  OrdersIcon,
+  ProductsIcons,
+  TableEditIcon,
+  TranslationsIcon,
+  VouchersIcon,
+} from "@saleor/macaw-ui/next";
+import React from "react";
+import { IntlShape } from "react-intl";
+
+import { getMenuItemExtension, mapToExtensionsItems } from "./legacy/utils";
+
+export interface FilterableMenuItem extends Omit<any, "children"> {
```

Comment:
```suggestion
export interface FilterableMenuItem extends Omit<SidebarMenuItem, "children"> {
```

Reply:
This is still WIP so I've decided to loosen it a bit of type-checking for now as it will change dynamically but before final release, I plan to add strict type checking so not worry 😄